src/webots/nodes/WbPbrAppearance.cpp

```diff
@@ -595,3 +685,48 @@ bool WbPbrAppearance::exportNodeHeader(WbVrmlWriter &writer) const {
     return true;
   return WbAbstractAppearance::exportNodeHeader(writer);
 }
+
+void WbPbrAppearance::exportShallowNode(WbVrmlWriter &writer) const {
```

Comment:
I think it's better yes, because they have been instantiated "artificially" (i.e not from a tokenizer nor copied from an existing node) so most of the member variables of these nodes, for instance at the basenode level, are not defined. Hence although letting the "normal" writer do the export might technically be possible, it entails adding a ton of conditionals just about everywhere to prevent segmentation faults at every corner, which is not really a clean approach and definitely harder to maintain in the long term. Or at least, that was the rationale behind the choice